app/signin/_components/SignInForm.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+
+import { signin } from "~/actions/auth";
+import { Button } from "~/components/ui/button";
+import { Input } from "~/components/ui/input";
+import { Label } from "~/components/ui/label";
+import { useFormState } from "react-dom";
+import { useEffect } from "react";
+import { redirect } from "next/navigation";
+
+const SignInForm = () => {
+  const initialState = { error: null, success: false };
+  const [formState, formAction] = useFormState(signin, initialState);
+
+  useEffect(() => {
+    if (formState.success) {
+      // redirect to home page
+      redirect("/");
+    }
+  }, [formState]);
+
+  return (
+    <form action={formAction}>
+      {formState.error && (
+        <div className="text-red-500 text-center text-sm">
+          {formState.error}
```

Comment:
Same comment as `SignUpForm` - can this be `error.message`

Reply:
Answered it here: https://github.com/complexdatacollective/Studio/pull/13#discussion_r1631011774